### PR TITLE
Save occurence of anonymous class symbol

### PR DESCRIPTION
**Pull request by tgodzik**
Not entirely sure it's the way to go, but the occurrence is saved for sure, the only problem is that it looks to be saved as reference and not definition.

@olafurpg Is this the way to go? Will try to look at how to make it def tomorrow.

Edit: Ok, added it to `binders` set - that's seems to define what is a definition

### CODE REVIEW

File: tests/jvm/src/test/resources/example/Advanced.scala [hunk 1]
```diff
@@ -10,8 +10,8 @@ class C/*<=advanced.C#*/[T/*<=advanced.C#[T]*/] {
 
 class Structural/*<=advanced.Structural#*/ {
   def s1/*<=advanced.Structural#s1().*/: { val x/*<=local0*/: Int/*=>scala.Int#*/ } = ???/*=>scala.Predef.`???`().*/
-  def s2/*<=advanced.Structural#s2().*/ = new { val x/*<=local1*/: Int/*=>scala.Int#*/ = ???/*=>scala.Predef.`???`().*/ }
-  def s3/*<=advanced.Structural#s3().*/ = new { def m/*<=local3*/(x/*<=local4*/: Int/*=>scala.Int#*/): Int/*=>scala.Int#*/ = ???/*=>scala.Predef.`???`().*/ }
+  def s2/*<=advanced.Structural#s2().*/ = new { val x: Int = ??? }/*<=local2*/
```
Comment (tgodzik) [inline tests/jvm/src/test/resources/example/Advanced.scala]: Seems it ignores the stuff inside now... need to fix that somehow.

File: tests/jvm/src/test/resources/example/Traits.scala [hunk 2]
```diff
@@ -6,7 +6,7 @@ trait T/*<=traits.T#*/ {
 
 sealed trait U/*<=traits.U#*/
 object U/*<=traits.U.*/ {
-  def u/*<=traits.U.u().*/: U/*=>traits.U#*/ = new U/*=>traits.U#*/ /*=>java.lang.Object#`<init>`().*/{}
+  def u/*<=traits.U.u().*/: U/*=>traits.U#*/ = new U/*<=local0*/ /*=>java.lang.Object#`<init>`().*/{}
```
Comment (tgodzik) [inline tests/jvm/src/test/resources/example/Traits.scala]: Now it seems to override the occurrence of `U`

Comment (tgodzik) [reply tests/jvm/src/test/resources/example/Traits.scala]: Fixed that, but not 100% sure about the result :/

File: semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TextDocumentOps.scala [hunk 3]
```diff
@@ -203,6 +203,14 @@ trait TextDocumentOps { self: SemanticdbOps =>
 
             saveSymbol(gsym)
             if (gsym.isClass && !gsym.isTrait) {
+              if (gsym.isAnonymousClass) {
+                val pos = gsym.pos.toMeta
```
Comment (olafurpg) [inline semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TextDocumentOps.scala]: There is a “focus” method on compiler positions to make the start and end point to the same offset. That would avoid the need for startPosition below.

Comment (tgodzik) [reply semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TextDocumentOps.scala]: Fixed

**Pull request merged**